.devcontainer/devcontainer.json

```diff
@@ -21,20 +21,49 @@
 		"go.inferGopath": true,
 		"go.toolsGopath": "/root/go",
 		"[go]": {
+			"editor.formatOnSave": true,
+			"editor.codeActionsOnSave": {
+				"source.organizeImports": true,
+			},
+			// Optional: Disable snippets, as they conflict with completion ranking.
 			"editor.snippetSuggestions": "none",
+		},
+		"[go.mod]": {
 			"editor.formatOnSave": true,
 			"editor.codeActionsOnSave": {
-				"source.organizeImports": true
+				"source.organizeImports": true,
 			},
-			"editor.codeActionsOnSaveTimeout": 3000
 		},
 		"gopls": {
-			"usePlaceholders": true, // add parameter placeholders when completing a function
-			// Experimental settings
-			"completeUnimported": true, // autocomplete unimported packages
-			"watchFileChanges": true, // watch file changes outside of the editor
-			"deepCompletion": true // enable deep completion
+			// Add parameter placeholders when completing a function.
+			"usePlaceholders": true,
+
+			// If true, enable additional analyses with staticcheck.
+			// Warning: This will significantly increase memory usage.
+			"staticcheck": false,
+		},
+		"go.alternateTools": {
+			// "goimports": "gofumpt"
+		},
+		"go.buildFlags": [
+			"-v"
+		],
+		"go.testFlags": [
+			"-count=1"
+		],
+		"go.testTimeout": "90s",
+		"go.formatTool": "goimports",
+		"go.liveErrors": {
+			"enabled": true,
+			"delay": 500
 		},
+		// "go.lintOnSave": "workspace",
+		"go.lintOnSave": "package",
+		"go.lintTool": "golangci-lint",
+		"go.lintFlags": [
+			"--fast"
```

Comment:
Only once I received timeout error - cannot catch it again.
(longer `--timeout` as the future fix can help)